### DISCUSSION TITLE
Need help regarding the download excel file.

### DISCUSSION THREAD

**Discussion by ehtishamalli** [PHPOffice/PhpSpreadsheet#3949] [Q&A] [answered]
Hi,

I need a help regarding installing this library i install it with composer and when i try to download file with sample basic example [01_Simple_download_xlsx.php](https://github.com/PHPOffice/PhpSpreadsheet/blob/master/samples/Basic/01_Simple_download_xlsx.php)
it download the file.

but when I try to open file it give me an error of "The file format or extension doesn`t match provide valid extension or verify your file". and when i try to open it some random alphabets is written in file. can you help me with this.

thanks

<img width="677" alt="Screenshot 2024-03-16 173814" src="https://github.com/PHPOffice/PhpSpreadsheet/assets/154227711/4437a0f4-a465-488c-9270-221c7b2bc925">

<img width="734" alt="Screenshot 2024-03-16 174023242" src="https://github.com/PHPOffice/PhpSpreadsheet/assets/154227711/a515a29b-7c5f-4c5a-b942-6b503e0fc3e1">

**Comment by oleibman** [comment, marked as answer]
It appears that you downloaded the file, which should have extension `xlsx`, to a file with extension `xls`. If you rename the file to end with xlsx, or download it with that extension, you should have no trouble opening it.

**Comment by oleibman** [comment]
No update in over 5 months, no reason to think suggested answer is wrong. Marking it as answer.
